wasm/validator/derive/src/entrypoint.rs

```diff
@@ -140,11 +128,7 @@ fn impl_migrate_entrypoint(attr: &TokenStream, fn_item: syn::ItemFn) -> TokenStr
 
     assert!(
         matches!(sig.output, syn::ReturnType::Type(_, _)),
-        "Validator `permission_tokens()` entrypoint must have `MigrationResult` return type"
-    );
-    assert!(
-        attr.is_empty(),
-        "`#[entrypoint]` macro for Validator `migrate` entrypoint accepts no attributes"
+        "Validator `migrate()` entrypoint must have `MigrationResult` return type"
```

Comment:
One trick that you could do here is to insert the actual type with the fully qualified path. What this will do is also type check the output of the function.

Reply:
Closed in favour of #3764

Reply:
@appetrosyan , this is actually done in line 143